tests/run-make-fulldeps/staticlib-dylib-linkage/Makefile

```diff
@@ -0,0 +1,16 @@
+include ../tools.mk
+
+all:
+	$(RUSTC) -C prefer-dynamic bar.rs
+	$(RUSTC) foo.rs --crate-type staticlib --print native-static-libs \
+		-Z staticlib-allow-rdylib-deps 2>&1 | grep 'note: native-static-libs: ' \
+		| sed 's/note: native-static-libs: \(.*\)/\1/' > $(TMPDIR)/libs.txt
+	cat $(TMPDIR)/libs.txt
+
+ifdef IS_MSVC
+	$(CC) $(CFLAGS) foo.c $(TMPDIR)/foo.lib $$(cat $(TMPDIR)/libs.txt) $(call OUT_EXE,foo)
```

Comment:
Looks like for MSVC it will be necessary to split out the compiling of C code and linking as the C compiler doesn't accept linker args unlike for gcc and clang.

Reply:
Maybe [`/link`](https://learn.microsoft.com/en-us/cpp/build/reference/link-pass-options-to-linker?view=msvc-170) can be used to pass arguments to the linker?

Reply:
Maybe, but that did require splitting the individual arguments printed by `--print native-static-libs` and prepending each with `/link`, right? Separating the linker invocation is easier I think.

Reply:
The docs suggest that everything after `/link` will be passed to the linker, not just a single argument. I have switched the test to using `/link`.